File: frontend/web/services/workflows/get.ts
import { ServiceReturn } from "@/types/api";
import { axiosInstance } from "@/services";
import { SERVICE_ERROR_UNKNOWN } from "@/config/services";
import { Workflow } from "@/types/workflows";

export const getAll = async (): Promise<ServiceReturn<Workflow[]>> => {
	try {
		const response = await axiosInstance.get<Workflow[]>(`/workflows`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

export const getOne = async (workflowId: string): Promise<ServiceReturn<Workflow>> => {
	try {
		const response = await axiosInstance.get<Workflow>(`/workflows/${workflowId}`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

File: frontend/mobile/lib/services/workflows/get.dart
import 'package:area_mobile/services/dio.dart';
import 'package:area_mobile/types/services.dart';
import 'package:area_mobile/types/workflows/workflows.dart';

Future<ServiceReturn<List<Workflow>>> getAll() async {
  try {
    final response = await dio.get<List<dynamic>>('/workflows');
    print(response.data);
    return ServiceReturn(
        data: response.data!.map((e) => Workflow.fromJson(e)).toList());
  } catch (e) {
    print(e.toString());
    return ServiceReturn(error: e.toString());
  }
}

Future<ServiceReturn<Workflow>> getOne(String workflowId) async {
  try {
    final response = await dio.get<Workflow>('/workflows/$workflowId');
    return ServiceReturn(data: response.data);
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}
